homeassistant/components/gogogate2/__init__.py

```diff
@@ -1 +1,36 @@
 """The gogogate2 component."""
+from homeassistant.components.cover import DOMAIN as COVER_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .common import get_data_update_coordinator
+
+
+async def async_setup(hass: HomeAssistant, base_config: dict) -> bool:
+    """Set up for Gogogate2 controllers."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Do setup of Gogogate2."""
+    data_update_coordinator = get_data_update_coordinator(hass, config_entry)
+    await data_update_coordinator.async_refresh()
+
+    if not data_update_coordinator.last_update_success:
```

Comment:
Consider using `DataUpdateCoordinator` directly since the only thing being added is the `api` object like the example here
https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
`get_data_update_coordinator` is used in multiple places (__init__.py and in testing) to reduce duplicate code. I don't think I can create the update coordinator in __init__.py without introducing duplication.